frontend/src/metabase/browse/components/TableBrowser/TableBrowser.jsx

```diff
@@ -143,10 +141,6 @@ const TableBrowserItemButtons = ({ tableId, dbId, xraysEnabled }) => {
 
 TableBrowserItemButtons.propTypes = itemButtonsPropTypes;
 
-const isTableLoading = (table, database) => {
-  return database && isSyncInProgress(database) && isSyncInProgress(table);
```

Comment:
I'm not entirely sure why we need to remove the database status check here. I've read the spec, but couldn't make out why this is needed. Also, just to give you some context, I don't know much about how database or table's `initial_sync_status` are related i.e. if one's status influence another or how timing each of these status changes, maybe that's another reason why I don't understand this part. Could you help elaborate?

Reply:
Sure. Currently we display tables as syncing in `/browse` with loading spinners when the database is first added by an admin, and the table hasn't fully been synced yet. This PR aims to display newly added tables as syncing whenever we do a normal database sync.

From the spec:

> We have an [existing approach](https://www.notion.so/metabase/Provide-progress-on-initial-database-sync-16af8e79312a45ecb715d8e58801c52b?pvs=4#1d8f0cf533c1481d81a9e33021143e76) to this that we use for the initial database sync, where we show new tables that are syncing and grey them out until they are ready across different product areas. Let's use this to provide more visibility into new table syncs.

Note we have two slightly different meanings of initial_sync_status = "incomplete", for tables and databases:

`database.initial_sync_status = "incomplete"` when the database is first added, and has not finished syncing yet.
`table.initial_sync_status = "incomplete"` when the table is first added (or was previously inactive), and has not finished syncing yet.

So during initial sync:

1. A database is created, with `database.initial_sync_status = "incomplete"`
2. Tables are created, with `table.initial_sync_status = "incomplete"`
3. One by one, tables complete syncing and have `table.initial_sync_status = "complete"`
4. After all tables are finished, we set `database.initial_sync_status = "complete"` 

Subsequent syncs:
1. `database.initial_sync_status = "complete"` the whole time
2. New tables are created, with `table.initial_sync_status = "incomplete"`
2. One by one, tables complete syncing and have `table.initial_sync_status = "complete"`

Reply:
Ok, this makes perfect sense. Thank you for the thorough explanation 👍